Mixpanel/MixpanelManager.cs

```diff
@@ -0,0 +1,182 @@
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+using mixpanel.queue;
+using UnityEngine;
+using UnityEngine.Networking;
+
+namespace mixpanel
+{
+    internal class MixpanelManager : MonoBehaviour
+    {
+        private const string TrackUrl = "https://api.mixpanel.com/track/?ip=1";
+        private const string EngageUrl = "https://api.mixpanel.com/engage/?ip=1";
+
+        private const int BatchSize = 50;
+        
+        private List<Value> TrackQueue = new List<Value>(500);
+        private List<Value> EngageQueue = new List<Value>(500);
+        
+        #region Singleton
+        
+        private static MixpanelManager _instance;
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.BeforeSceneLoad)]
+        private static void InitializeBeforeSceneLoad()
+        {
+            _instance = new GameObject("Mixpanel").AddComponent<MixpanelManager>();
+            Debug.Log($"[Mixpanel] Track Queue Depth: {Mixpanel.TrackQueue.CurrentCountOfItemsInQueue}");
+            Debug.Log($"[Mixpanel] Engage Queue Depth: {Mixpanel.EngageQueue.CurrentCountOfItemsInQueue}");
+        }
+
+        [RuntimeInitializeOnLoadMethod(RuntimeInitializeLoadType.AfterSceneLoad)]
+        private static void InitializeAfterSceneLoad()
+        {
+            Mixpanel.CollectAutoProperties();
+        }
+        
+        #endregion
+
+        private bool _isBlocking;
+        private bool _needsFlush;
+        private const int RetryMaxTries = 10;
+
+        private IEnumerator Start()
+        {
+            DontDestroyOnLoad(this);
+            StartCoroutine(PopulatePools());
+            StartCoroutine(TrimQueues());
+            while (true)
+            {
+                yield return new WaitForSecondsRealtime(MixpanelSettings.Instance.FlushInterval);
+                Mixpanel.Flush();
+            }
+        }
+        
+        private IEnumerator PopulatePools()
+        {
+            for (int i = 0; i < 50; i++)
```

Comment:
This populates the object and array MixpanelValue pools with 1000 objects which in some performance testing I did helped with performance.  I've set the magic numbers to be new constant variables PoolFillFrames and PoolFillEachFrame to remove the magic numbers and make it more descriptive.  The reason i do it with 2 numbers is to slowly load the pools over the first N frames of the unity app starting up because creating the MixpanelValue objects can be intensive so this spreads the load out over N frames and then once the pools are populated the internals of the library reuse these objects to keep GC and Memory allocations down.  Which improves performance.  Does this work for you?